### PR TITLE
remove arm endpoint, subscription id from required parameter. Retrieve them from context.

### PR DESCRIPTION
**Pull request by yunzhangMS**

### CODE REVIEW

File: Syndication/AzureStack.MarketplaceSyndication.psm1 [hunk 1]
```diff
@@ -560,6 +556,7 @@ function Import-AzSOfflineMarketplaceItem
     $ctx = Get-AzureRmContext
     $AccessToken = Resolve-AccessToken -Context $ctx -AccessToken $AccessToken
     $headers = @{ 'authorization' = "Bearer $AccessToken"}
+    $ArmEndpoint = $ctx.Environment.ResourceManagerUrl
```
Comment (serpop) [inline Syndication/AzureStack.MarketplaceSyndication.psm1]: $ctx and $headers starts with lower-case letter, $AccessToken and $ArmEndpoint starts with upper case letter.. Can you please be consistent in coding style?

File: Syndication/AzureStack.MarketplaceSyndication.psm1 [hunk 2]
```diff
@@ -692,18 +689,15 @@ function Import-ByDependency
...
-
-        [parameter(mandatory = $true)]
-        [String] $SubscriptionId
+        [String] $Destination
     )
 
     $ctx = Get-AzureRmContext
     $AccessToken = Resolve-AccessToken -Context $ctx -AccessToken $AccessToken
     $headers = @{ 'authorization' = "Bearer $AccessToken"}
+    $ArmEndpoint = $ctx.Environment.ResourceManagerUrl
```
Comment (serpop) [inline Syndication/AzureStack.MarketplaceSyndication.psm1]: Same here. You can uither use camel casing or pascal casing, but please do not mix them in one script.

File: Syndication/AzureStack.MarketplaceSyndication.psm1 [hunk 3]
```diff
@@ -692,18 +689,15 @@ function Import-ByDependency
...
+    $defaultProviderSubscription = Get-AzureRmSubscription -SubscriptionName 'Default Provider Subscription' | Select-AzureRmSubscription
+    $subscriptionId = $defaultProviderSubscription.subscription.id
```
Comment (serpop) [inline Syndication/AzureStack.MarketplaceSyndication.psm1]: This code looks weird and it takes some time to understand what does it do. I would recommend split Get- and Select- statements.

File: Syndication/AzureStack.MarketplaceSyndication.psm1 [hunk 4]
```diff
@@ -692,18 +689,15 @@ function Import-ByDependency
...
 
     $dirs = Get-ChildItem -Path $Destination
```
Comment (serpop) [inline Syndication/AzureStack.MarketplaceSyndication.psm1]: Are you sure it's always a directory?

File: Syndication/AzureStack.MarketplaceSyndication.psm1 (hunk 3 above)
Comment (yunzhangMS) [reply Syndication/AzureStack.MarketplaceSyndication.psm1]: fixed, thanks

File: Syndication/AzureStack.MarketplaceSyndication.psm1 (hunk 4 above)
Comment (yunzhangMS) [reply Syndication/AzureStack.MarketplaceSyndication.psm1]: fixed, thanks

**Pull request merged**